src/main/java/teammates/ui/webapi/GetFeedbackSessionLogsAction.java

```diff
@@ -30,7 +49,74 @@ void checkSpecificAccessControl() {
     }
 
     @Override
-    ActionResult execute() {
-        return null;
+    JsonResult execute() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        if (logic.getCourse(courseId) == null) {
+            return new JsonResult("Course not found", HttpStatus.SC_NOT_FOUND);
+        }
+        String email = getRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+        if (email != null && logic.getStudentForEmail(courseId, email) == null) {
+            return new JsonResult("Student not found", HttpStatus.SC_NOT_FOUND);
+        }
+        String startTimeStr = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_LOG_STARTTIME);
+        String endTimeStr = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_LOG_ENDTIME);
+        Instant startTime = Instant.ofEpochMilli(Long.parseLong(startTimeStr));
+        Instant endTime = Instant.ofEpochMilli(Long.parseLong(endTimeStr));
+        // TODO: we might want to impose limits on the time range from startTime to endTime
+
+        try {
+            List<FeedbackSessionLogEntry> fsLogEntries = logic.getFeedbackSessionLogs(courseId, email, startTime, endTime);
+            FeedbackSessionLogsData fslData = createFeedbackSessionLogsData(courseId, fsLogEntries);
+            return new JsonResult(fslData);
+        } catch (LogsServiceException e) {
+            return new JsonResult(e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    private FeedbackSessionLogsData createFeedbackSessionLogsData(String courseId,
```

Comment:
whats all this from this line onwards? Looks like a lot of code to convert between 2 data formats... you need to create a constructor in your output class that takes in the Data Transfer Object

Reply:
Noted!